server/utils/nlptodata.js

```diff
@@ -4,93 +4,123 @@ var stringSimilarity = require('string-similarity');
+                if(entityKeys.includes('condition_y')){
+                    columnname.push(entities.condition_y);
+                }
+                if(entityKeys.includes('condition_const_y')){
+                    columnname.push(entities.condition_const_y);
+                }
+                if(entityKeys.includes('z')){
+                    var matche_z = stringSimilarity.findBestMatch(entities.z, actualColumnName).bestMatch.target;
+                    columnname.push(matche_z);
+                }
+                if(entityKeys.includes('operation')){
+                    columnname.push(entities.operation);
+                }
+                if(entityKeys.includes('condition')){
+                    columnname.push(entities.condition);
+                }
+                if(entityKeys.includes('condition_const')){
+                    columnname.push(entities.condition_const);
+                }
 
-            if(isGrouped){
-                var dfx = df.groupBy(function (row) {
-                    //consolr.log(row)
-                    return row[entities.y];
-                });
-            }
+                /*
+                //var csvOptions = { columnNames:  actualColumnName};
+                //var df = DF.readFileSync(csvFilePath).parseCSV(csvOptions);//.toArray();
+                //console.log('entities', entities.x);
+                //console.log(df.first()['Employee Name']);
+                
+                var iterator = df.getIterator();
+                while (iterator.moveNext()) {
+                    var pair = iterator.getCurrent();
+                    //console.log(pair[1]);
+                }
+                
+                //console.log(matche_x, matche_y);
+                var df_sub = df.subset([matche_x, matche_y]);
+                
+                var data = formatting(df_sub);
+                data["plot_type"] = entities.plot_type;
+                data["user_query"] = nlpobj.text;
+                //console.log(data);
 
-            if(isCount){
+                if(isGrouped){
+                    var dfx = df.groupBy(function (row) {
+                        //consolr.log(row)
+                        return row[entities.y];
+                    });
+                }
 
-            }
-            fulfill(data);
-        } else {
-            reject("Sorry, your query is meaningless to us.");
-        }
-    });
-}
+                if(isCount){
 
-var formatting = function(df){
+                }
+                */
 
-    var columns = df.getColumns();
-    var arrayOfColumns = columns.toArray();
-    data = {};
-    for (var column in arrayOfColumns) {
-        var name = arrayOfColumns[column].name;
-        data[name] = [];
-        var series = arrayOfColumns[column].series;
-        series.forEach(function (value) {
-            data[name].push(value);
+                fulfill(data);
+            } else {
+                reject("Sorry, your query is meaningless to us.");
```

Comment:
sounds harsh, but its fine 😄

Reply:
haaaa haa..  You can change it to something softer.